Tests/SonarRunner.Shim.Tests/PropertiesWriterTest.cs

```diff
@@ -125,6 +127,39 @@ public void PropertiesWriterToString()
         }
 
         [TestMethod]
+        public void PropertiesWriter_ComputeProjectBaseDir()
+        {
+            AssertHasProjectBaseDir(new List<string> { @"c:\src\A\1.proj" }, @"c:\src\A", @"z:\");
+            AssertHasProjectBaseDir(new List<string> { @"c:\src\A\1.proj", @"c:\src\A\2.proj" }, @"c:\src\A", @"z:\");
```

Comment:
Perhaps add a test case to check that non-canonical paths are handled correctly e.g. "c:\src\A\dummy\..\.\1.proj" (I think they should be, as I think GetProjectDirectory returns the dir in a canonical form, but it wouldn't hurt to add a check just in case that changes).